sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -63,7 +63,10 @@ case class CreateFunctionCommand(
     } else {
       // For a permanent, we will store the metadata into underlying external catalog.
       // This function will be loaded into the FunctionRegistry when a query uses it.
-      // We do not load it into FunctionRegistry right now.
+      // We do not load it into FunctionRegistry right now. However, we need to validate.
+      catalog.loadFunctionResources(resources)
+      catalog.makeFunctionBuilder(functionName, className)
```

Comment:
I did a try in my local environment. It works well if we register it to `FunctionRegistry ` directly. I did not see the reason why we just validate it only. That is why I think this is not in a right direction.